buildSrc/src/main/groovy/org/edx/builder/ConfigHelper.groovy

```diff
@@ -95,4 +95,49 @@ class ConfigHelper {
             it.write(new JsonBuilder(configJson).toPrettyString())
         }
     }
+
+    def generateGoogleServicesJson(applicationId) {
+        def config = fetchConfig()
+        def firebase = config.get("FIREBASE")
+        if (!firebase) {
+            return
+        }
+        if (!firebase.getOrDefault("ENABLED", false)) {
+            return
+        }
+
+        def googleServicesJsonPath = projectDir.path + "/app/"
+        new File(googleServicesJsonPath).mkdirs()
+
+        def projectInfo = [
+                project_number: firebase.getOrDefault("PROJECT_NUMBER", ""),
+                project_id    : firebase.getOrDefault("PROJECT_ID", ""),
+                storage_bucket: "${firebase.getOrDefault("PROJECT_ID", "")}.appspot.com"
+        ]
+        def clientInfo = [
+                mobilesdk_app_id   : firebase.getOrDefault("APPLICATION_ID", ""),
+                android_client_info: [
+                        package_name: applicationId
+                ]
+        ]
+        def client = [
+                client_info : clientInfo,
+                oauth_client: [],
```

Comment:
This PR concentrates on moving FirebaseOptions configuring to the google-services.json.
```
val options = FirebaseOptions.Builder()
                .setProjectId(firebaseConfig.projectId)
                .setApplicationId(firebaseConfig.applicationId)
                .setApiKey(firebaseConfig.apiKey)
                .setGcmSenderId(firebaseConfig.gcmSenderId)
                .build()
```

Reply:
I think it's better to replicate the complete `google-services.json` as part of this PR.

Also, this may create the tracing issue if any new Firebase service doesn't work with the current properties of `google-services.json`

Reply:
> I think it's better to replicate the complete `google-services.json` as part of this PR.

This is not the focus of this PR.
The current implementation presented in this PR is sufficient to cover the migration from the previous approach of manual configuring.